Userland/Libraries/LibWeb/Layout/InlineFormattingContext.cpp

```diff
@@ -234,24 +234,74 @@ void InlineFormattingContext::apply_justification_to_fragments(CSS::TextJustify
     }
 }
 
+class InlineLevelLineWrappingIterator {
+public:
+    InlineLevelLineWrappingIterator(InlineLevelIterator& iterator)
+        : m_iterator(iterator)
+    {
+    }
+
+    Optional<InlineLevelIterator::Item> next()
+    {
+        if (m_next_items.is_empty())
+            return m_iterator.next();
+        return m_next_items.dequeue();
+    }
+
+    CSSPixels next_width()
```

Comment:
It's not clear from the name what this function does. Can we call it something more descriptive?